app/controllers/api/v1/taxonomies_controller.rb

```diff
@@ -3,13 +3,14 @@ class API::V1::TaxonomiesController < ApplicationController
   
     def index
 
-        if params[:directory].present?
+        if params[:directory].present? && APP_CONFIG['directories'].present?
 
             # scout sends through lowercase label 'bfis', 'bod' etc - look up the name in app config to send to the application
             @get_value_from_label = APP_CONFIG['directories'].find{|directory| directory["label"] === params[:directory]};
+            @results = (!@get_value_from_label.nil?) ? Taxonomy.filter_by_directory(@get_value_from_label["value"]) : {};
```

Comment:
I don't think you need this check here - if you do `Taxonomy.filter_by_directory(nil)` it just returns an empty collection.

So this could just be: ` @results = Taxonomy.filter_by_directory(@get_value_from_label["value"])`

Reply:
Also btw - if you just want to create a variable and you don't need to expose it somewhere else as an instance variable (for example in the views), you don't need to use the @, you can just do:

```
results = Taxonomy.filter_by_directory(get_value_from_label["value"])
```

Means it's private to this method but that's ok as we don't need these vars elsewhere 😄

Reply:
So we do still need some kind of switch in here because `get_value_from_label` returns {"value"=>"Buckinghamshire Online Directory", "label"=>"bod"} and it was causing trouble when I did get_value_from_label["value"] so I've updated it a bit in the new code - feels less clunky now! 🤞

Reply:
Ahhh gotcha! I think we can do the following in that case:

```
value = APP_CONFIG['directories'].find{|directory| directory["label"] === params[:directory]}&.fetch('value')
results = Taxonomy.filter_by_directory(value)
```
Chaining with & and `fetch` means it won't error if the directory is nil, it will just return nil.